contributed/matching_fof_mlz_blending.ipynb

```diff
@@ -0,0 +1,805 @@
+    "plt.xlim(-0.5,5.5)\n",
+    "plt.ylim(-0.5,5.5)\n",
+    "plt.colorbar(label=r'$\\log(N_{\\rm groups} \\, + \\, 1)$');"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Let's further inspect the objects in the groups that have 1-to-1 truth/coadd match.\n",
+    "\n",
+    "# first, let's find our the IDs of the groups that have 1-to-1 truth/coadd match:\n",
+    "one_to_one_group_mask = np.in1d(results['group_id'], np.flatnonzero((n_truth == 1) & (n_coadd == 1)))\n",
+    "\n",
+    "# and then we can find the row indices in the *original* truth/coadd catalogs for those 1-to-1 groups\n",
+    "truth_idx = results['row_index'][one_to_one_group_mask & truth_mask]\n",
+    "coadd_idx = results['row_index'][one_to_one_group_mask & coadd_mask]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Let's further inspect the objects in the groups that have 2-to-1 truth/coadd match.\n",
+    "# This group will contain possible blended objects.\n",
+    "# first, let's find our the IDs of the groups that have 2-to-1 truth/coadd match:\n",
+    "two_to_one_group_mask = np.in1d(results['group_id'], np.flatnonzero((n_truth == 2) & (n_coadd == 1)))\n",
+    "\n",
+    "# and then we can find the row indices in the *original* truth/coadd catalogs for those 2-to-1 groups\n",
+    "truth_idx21 = results['row_index'][two_to_one_group_mask & truth_mask]\n",
+    "coadd_idx21 = results['row_index'][two_to_one_group_mask & coadd_mask]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Let's further inspect the objects in the groups that have 1-to-2 truth/coadd match.\n",
+    "# This group will contain mis-detected objects.\n",
+    "# first, let's find our the IDs of the groups that have 2-to-1 truth/coadd match:\n",
+    "one_to_two_group_mask = np.in1d(results['group_id'], np.flatnonzero((n_truth == 1) & (n_coadd == 2)))\n",
+    "\n",
+    "# and then we can find the row indices in the *original* truth/coadd catalogs for those 1-to-2 groups\n",
+    "truth_idx12 = results['row_index'][one_to_two_group_mask & truth_mask]\n",
+    "coadd_idx12 = results['row_index'][one_to_two_group_mask & coadd_mask]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Check number of objects in each group.\n",
+    "print (len(truth_idx21))\n",
+    "print (len(coadd_idx21))\n",
+    "print (len(coadd_idx))\n",
+    "print (len(truth_idx12))\n",
+    "print (len(coadd_idx12))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Use truth catalog's spec-z and magitudes to generate train and test catalogs for MLZ.\n",
+    "truth_z=truth_data['redshift'][truth_idx]\n",
```

Comment:
Just a suggestion that you can turn the coadd and truth catalogs into Pandas DataFrames, in which case you can do all this in one line and reduce the risk of typos! Like, `dataframe.loc[mask, [list of columns]]`. This applies to all 4 sets.